isisd/isis_pfpacket.c

```diff
@@ -277,19 +277,41 @@ int isis_recv_pdu_bcast(struct isis_circuit *circuit, uint8_t *ssnpa)
 		return ISIS_WARNING;
 	}
 
-	/* on lan we have to read to the static buff first */
-	bytesread = recvfrom(circuit->fd, sock_buff, sizeof(sock_buff),
-			     MSG_DONTWAIT, (struct sockaddr *)&s_addr,
-			     (socklen_t *)&addr_len);
-	if (bytesread < 0) {
-		zlog_warn("isis_recv_pdu_bcast(): recvfrom() failed");
-		return ISIS_WARNING;
-	}
-
-	/* then we lose the LLC */
-	stream_write(circuit->rcv_stream, sock_buff + LLC_LEN,
-		     bytesread - LLC_LEN);
+	/* on lan we have to read to the static buff first; however, that is
+	 * hard-coded to be of size 8192. If the mtu of the circuit is greater
+	 * than that and hello padding is enabled on this interface, we will
+	 * cut the packet short, and later on pdu_len_validate will fail. To be
+	 * on the safe side, let's ensure that we have enough space
+	 */
+	unsigned int max_size =
+		circuit->interface->mtu > circuit->interface->mtu6
+			? circuit->interface->mtu
+			: circuit->interface->mtu6;
+	if (max_size > sizeof(sock_buff)) {
```

Comment:
If we can use a stack buffer do we need the static buffer at all? Would certainly simplify this code